diff --git a/external/storm-kafka-client/src/main/java/org/apache/storm/kafka/trident/TridentKafkaState.java b/external/storm-kafka-client/src/main/java/org/apache/storm/kafka/trident/TridentKafkaState.java
index f24fed597..f93951925 100644
--- a/external/storm-kafka-client/src/main/java/org/apache/storm/kafka/trident/TridentKafkaState.java
+++ b/external/storm-kafka-client/src/main/java/org/apache/storm/kafka/trident/TridentKafkaState.java
@@ -79,12 +79,13 @@ public class TridentKafkaState implements State {
             for (TridentTuple tuple : tuples) {
                 topic = topicSelector.getTopic(tuple);
 
+                Object keyFromTuple = mapper.getKeyFromTuple(tuple);
                 if(topic != null) {
                     Future<RecordMetadata> result = producer.send(new ProducerRecord(topic,
-                            mapper.getKeyFromTuple(tuple), mapper.getMessageFromTuple(tuple)));
+                            keyFromTuple, mapper.getMessageFromTuple(tuple)));
                     futures.add(result);
                 } else {
-                    LOG.warn("skipping key = " + mapper.getKeyFromTuple(tuple) + ", topic selector returned null.");
+                    LOG.warn("skipping key = " + keyFromTuple + ", topic selector returned null.");
                 }
             }
 
