diff --git a/storm-core/src/jvm/backtype/storm/messaging/netty/Client.java b/storm-core/src/jvm/backtype/storm/messaging/netty/Client.java
index f332bb3..2e036af 100644
--- a/storm-core/src/jvm/backtype/storm/messaging/netty/Client.java
+++ b/storm-core/src/jvm/backtype/storm/messaging/netty/Client.java
@@ -547,8 +547,9 @@
     }
 
     private synchronized void closeChannel() {
-        if (channelRef.get() != null) {
-            channelRef.get().close();
+        Object channel = channelRef.get();
+		if (channel != null) {
+            channel.close();
             LOG.debug("channel to {} closed", dstAddressPrefixedName);
         }
     }
