diff --git a/src/main/java/org/apache/commons/codec/language/bm/Rule.java b/src/main/java/org/apache/commons/codec/language/bm/Rule.java
index 644614a..dcd3759 100644
--- a/src/main/java/org/apache/commons/codec/language/bm/Rule.java
+++ b/src/main/java/org/apache/commons/codec/language/bm/Rule.java
@@ -84,8 +84,9 @@
         public static final Comparator<Phoneme> COMPARATOR = new Comparator<Phoneme>() {
             @Override
             public int compare(final Phoneme o1, final Phoneme o2) {
-                for (int i = 0; i < o1.phonemeText.length(); i++) {
-                    if (i >= o2.phonemeText.length()) {
+                int o2Length = o2.phonemeText.length();
+				for (int i = 0; i < o1.phonemeText.length(); i++) {
+                    if (i >= o2Length) {
                         return +1;
                     }
                     final int c = o1.phonemeText.charAt(i) - o2.phonemeText.charAt(i);
@@ -94,7 +95,7 @@
                     }
                 }
 
-                if (o1.phonemeText.length() < o2.phonemeText.length()) {
+                if (o1.phonemeText.length() < o2Length) {
                     return -1;
                 }
 
