diff --git a/src/main/java/org/apache/commons/math3/linear/RealVector.java b/src/main/java/org/apache/commons/math3/linear/RealVector.java
index 89611c34c..499e68407 100644
--- a/src/main/java/org/apache/commons/math3/linear/RealVector.java
+++ b/src/main/java/org/apache/commons/math3/linear/RealVector.java
@@ -643,11 +643,12 @@ public abstract class RealVector {
      */
     public RealMatrix outerProduct(RealVector v) {
         RealMatrix product;
+        int m = this.getDimension();
         if (v instanceof SparseRealVector || this instanceof SparseRealVector) {
-            product = new OpenMapRealMatrix(this.getDimension(),
+            product = new OpenMapRealMatrix(m,
                                             v.getDimension());
         } else {
-            product = new Array2DRowRealMatrix(this.getDimension(),
+            product = new Array2DRowRealMatrix(m,
                                                v.getDimension());
         }
         Iterator<Entry> thisIt = sparseIterator();
