diff --git a/src/java/org/apache/commons/math/linear/decomposition/LUDecompositionImpl.java b/src/java/org/apache/commons/math/linear/decomposition/LUDecompositionImpl.java
index c55a0fcf1..1d8ad1f60 100644
--- a/src/java/org/apache/commons/math/linear/decomposition/LUDecompositionImpl.java
+++ b/src/java/org/apache/commons/math/linear/decomposition/LUDecompositionImpl.java
@@ -332,7 +332,8 @@ public class LUDecompositionImpl implements LUDecomposition {
                 // Solve LY = b
                 for (int col = 0; col < m; col++) {
                     for (int i = col + 1; i < m; i++) {
-                        bp[i] -= bp[col] * lu[i][col];
+                        double bpCol = bp[col];
+                        bp[i] -= bpCol * lu[i][col];
                     }
                 }
 
