diff --git a/src/java/org/apache/commons/math/analysis/polynomials/PolynomialFunction.java b/src/java/org/apache/commons/math/analysis/polynomials/PolynomialFunction.java
index 2db950700..a7a7a6f3e 100644
--- a/src/java/org/apache/commons/math/analysis/polynomials/PolynomialFunction.java
+++ b/src/java/org/apache/commons/math/analysis/polynomials/PolynomialFunction.java
@@ -56,11 +56,12 @@ public class PolynomialFunction implements DifferentiableUnivariateRealFunction,
      */
     public PolynomialFunction(double c[]) {
         super();
-        if (c.length < 1) {
+        int l = c.length;
+        if (l < 1) {
             throw new IllegalArgumentException("Polynomial coefficient array must have postive length.");
         }
-        this.coefficients = new double[c.length];
-        System.arraycopy(c, 0, this.coefficients, 0, c.length);
+        this.coefficients = new double[l];
+        System.arraycopy(c, 0, this.coefficients, 0, l);
     }
 
     /**
