diff --git a/src/java/org/apache/commons/math/linear/BigMatrixImpl.java b/src/java/org/apache/commons/math/linear/BigMatrixImpl.java
index ccb9a9d14..925633231 100644
--- a/src/java/org/apache/commons/math/linear/BigMatrixImpl.java
+++ b/src/java/org/apache/commons/math/linear/BigMatrixImpl.java
@@ -1053,11 +1053,12 @@ public class BigMatrixImpl implements BigMatrix, Serializable {
             int max = col; // permutation row
             BigDecimal largest = ZERO;
             for (int row = col; row < nRows; row++) {
-                sum = lu[row][col];
+                BigDecimal[] luRow = lu[row];
+                sum = luRow[col];
                 for (int i = 0; i < col; i++) {
-                    sum = sum.subtract(lu[row][i].multiply(lu[i][col]));
+                    sum = sum.subtract(luRow[i].multiply(lu[i][col]));
                 }
-                lu[row][col] = sum;
+                luRow[col] = sum;
                 
                 // maintain best permutation choice
                 if (sum.abs().compareTo(largest) == 1) {
