diff --git a/src/java/org/apache/commons/math/linear/BiDiagonalTransformer.java b/src/java/org/apache/commons/math/linear/BiDiagonalTransformer.java
index 38131761a..4b1443b32 100644
--- a/src/java/org/apache/commons/math/linear/BiDiagonalTransformer.java
+++ b/src/java/org/apache/commons/math/linear/BiDiagonalTransformer.java
@@ -148,12 +148,13 @@ class BiDiagonalTransformer implements Serializable {
             final int n = householderVectors[0].length;
             double[][] bData = new double[m][n];
             for (int i = 0; i < main.length; ++i) {
-                bData[i][i] = main[i];
+                double[] bDataI = bData[i];
+                bDataI[i] = main[i];
                 if (i < main.length - 1) {
                     if (m < n) {
                         bData[i + 1][i] = secondary[i];
                     } else {
-                        bData[i][i + 1] = secondary[i];
+                        bDataI[i + 1] = secondary[i];
                     }
                 }
             }
