diff --git a/src/main/java/org/joda/time/tz/ZoneInfoCompiler.java b/src/main/java/org/joda/time/tz/ZoneInfoCompiler.java
index b03549f..28c6303 100644
--- a/src/main/java/org/joda/time/tz/ZoneInfoCompiler.java
+++ b/src/main/java/org/joda/time/tz/ZoneInfoCompiler.java
@@ -493,7 +493,8 @@
             }
 
             if (st.hasMoreTokens()) {
-                String token = st.nextToken();
+                String alias = st.nextToken();
+				String token = alias;
                 if (token.equalsIgnoreCase("Rule")) {
                     Rule r = new Rule(st);
                     RuleSet rs = iRuleSets.get(r.iName);
@@ -506,8 +507,8 @@
                 } else if (token.equalsIgnoreCase("Zone")) {
                     zone = new Zone(st);
                 } else if (token.equalsIgnoreCase("Link")) {
-                    iLinks.add(st.nextToken());
-                    iLinks.add(st.nextToken());
+                    iLinks.add(alias);
+                    iLinks.add(alias);
                 } else {
                     System.out.println("Unknown line: " + line);
                 }
