diff --git a/src/main/java/org/jsoup/parser/HtmlTreeBuilder.java b/src/main/java/org/jsoup/parser/HtmlTreeBuilder.java
index b8cc1f03..07be18e8 100644
--- a/src/main/java/org/jsoup/parser/HtmlTreeBuilder.java
+++ b/src/main/java/org/jsoup/parser/HtmlTreeBuilder.java
@@ -256,10 +256,11 @@ public class HtmlTreeBuilder extends TreeBuilder {
         Node node;
         // characters in script and style go in as datanodes, not text nodes
         String tagName = currentElement().tagName();
+        String data = characterToken.getData();
         if (tagName.equals("script") || tagName.equals("style"))
-            node = new DataNode(characterToken.getData());
+            node = new DataNode(data);
         else
-            node = new TextNode(characterToken.getData());
+            node = new TextNode(data);
         currentElement().appendChild(node); // doesn't use insertNode, because we don't foster these; and will always have a stack.
     }
 
