diff --git a/src/main/java/org/apache/commons/compress/archivers/zip/ZipArchiveInputStream.java b/src/main/java/org/apache/commons/compress/archivers/zip/ZipArchiveInputStream.java
index 32bb562a..d5934bc2 100644
--- a/src/main/java/org/apache/commons/compress/archivers/zip/ZipArchiveInputStream.java
+++ b/src/main/java/org/apache/commons/compress/archivers/zip/ZipArchiveInputStream.java
@@ -872,7 +872,8 @@ private void readDataDescriptor() throws IOException {
         final ZipLong potentialSig = new ZipLong(twoDwordBuf, DWORD);
         if (potentialSig.equals(ZipLong.CFH_SIG) || potentialSig.equals(ZipLong.LFH_SIG)) {
             pushback(twoDwordBuf, DWORD, DWORD);
-            current.entry.setCompressedSize(ZipLong.getValue(twoDwordBuf));
+            long size = ZipLong.getValue(twoDwordBuf);
+            current.entry.setCompressedSize(size);
             current.entry.setSize(ZipLong.getValue(twoDwordBuf, WORD));
         } else {
             current.entry.setCompressedSize(ZipEightByteInteger.getLongValue(twoDwordBuf));
