diff --git a/src/main/java/org/apache/commons/compress/compressors/lz4/FramedLZ4CompressorOutputStream.java b/src/main/java/org/apache/commons/compress/compressors/lz4/FramedLZ4CompressorOutputStream.java
index 81a17704..0536428e 100644
--- a/src/main/java/org/apache/commons/compress/compressors/lz4/FramedLZ4CompressorOutputStream.java
+++ b/src/main/java/org/apache/commons/compress/compressors/lz4/FramedLZ4CompressorOutputStream.java
@@ -210,13 +210,14 @@ public void write(final byte[] data, int off, int len) throws IOException {
         if (params.withContentChecksum) {
             contentHash.update(data, off, len);
         }
-        if (currentIndex + len > blockData.length) {
+        int blockDataLength = blockData.length;
+        if (currentIndex + len > length) {
             flushBlock();
-            while (len > blockData.length) {
-                System.arraycopy(data, off, blockData, 0, blockData.length);
-                off += blockData.length;
-                len -= blockData.length;
-                currentIndex = blockData.length;
+            while (len > blockDataLength) {
+                System.arraycopy(data, off, blockData, 0, blockDataLength);
+                off += blockDataLength;
+                len -= blockDataLength;
+                currentIndex = blockDataLength;
                 flushBlock();
             }
         }
