diff --git a/src/main/java/org/apache/commons/compress/archivers/sevenz/SevenZFile.java b/src/main/java/org/apache/commons/compress/archivers/sevenz/SevenZFile.java
index 5624b4c1..a86a05f2 100644
--- a/src/main/java/org/apache/commons/compress/archivers/sevenz/SevenZFile.java
+++ b/src/main/java/org/apache/commons/compress/archivers/sevenz/SevenZFile.java
@@ -979,7 +979,8 @@ private void readFilesInfo(final ByteBuffer header, final Archive archive) throw
                     header.get(names);
                     int nextFile = 0;
                     int nextName = 0;
-                    for (int i = 0; i < names.length; i += 2) {
+                    int namesLength = names.length;
+                    for (int i = 0; i < namesLength; i += 2) {
                         if (names[i] == 0 && names[i + 1] == 0) {
                             checkEntryIsInitialized(fileMap, nextFile);
                             fileMap.get(nextFile).setName(new String(names, nextName, i - nextName, StandardCharsets.UTF_16LE));
@@ -987,7 +988,7 @@ private void readFilesInfo(final ByteBuffer header, final Archive archive) throw
                             nextFile++;
                         }
                     }
-                    if (nextName != names.length || nextFile != numFiles) {
+                    if (nextName != namesLength || nextFile != numFiles) {
                         throw new IOException("Error parsing file names");
                     }
                     break;
