diff --git a/src/main/java/org/apache/commons/compress/archivers/zip/ZipArchiveOutputStream.java b/src/main/java/org/apache/commons/compress/archivers/zip/ZipArchiveOutputStream.java
index 02f0cf1c..b1008f35 100644
--- a/src/main/java/org/apache/commons/compress/archivers/zip/ZipArchiveOutputStream.java
+++ b/src/main/java/org/apache/commons/compress/archivers/zip/ZipArchiveOutputStream.java
@@ -1389,7 +1389,8 @@ protected void writeCentralFileHeader(final ZipArchiveEntry ze) throws IOExcepti
         final ByteBuffer commentB = getEntryEncoding(ze).encode(comm);
         final int nameLen = name.limit() - name.position();
         final int commentLen = commentB.limit() - commentB.position();
-        final int len= CFH_FILENAME_OFFSET + nameLen + extra.length + commentLen;
+        int extraLength = extra.length;
+        final int len= CFH_FILENAME_OFFSET + nameLen + extraLength + commentLen;
         final byte[] buf = new byte[len];
 
         System.arraycopy(CFH_SIG,  0, buf, CFH_SIG_OFFSET, WORD);
@@ -1429,7 +1430,7 @@ protected void writeCentralFileHeader(final ZipArchiveEntry ze) throws IOExcepti
         putShort(nameLen, buf, CFH_FILENAME_LENGTH_OFFSET);
 
         // extra field length
-        putShort(extra.length, buf, CFH_EXTRA_LENGTH_OFFSET);
+        putShort(extraLength, buf, CFH_EXTRA_LENGTH_OFFSET);
 
         putShort(commentLen, buf, CFH_COMMENT_LENGTH_OFFSET);
 
@@ -1461,9 +1462,9 @@ protected void writeCentralFileHeader(final ZipArchiveEntry ze) throws IOExcepti
         System.arraycopy(name.array(), name.arrayOffset(), buf, CFH_FILENAME_OFFSET, nameLen);
 
         final int extraStart = CFH_FILENAME_OFFSET + nameLen;
-        System.arraycopy(extra, 0, buf, extraStart, extra.length);
+        System.arraycopy(extra, 0, buf, extraStart, extraLength);
 
-        final int commentStart = extraStart + extra.length;
+        final int commentStart = extraStart + extraLength;
 
         // file comment
         System.arraycopy(commentB.array(), commentB.arrayOffset(), buf, commentStart, commentLen);
