diff --git a/src/main/java/org/apache/commons/compress/archivers/zip/ExtraFieldUtils.java b/src/main/java/org/apache/commons/compress/archivers/zip/ExtraFieldUtils.java
index dd22f038..20ac5d8b 100644
--- a/src/main/java/org/apache/commons/compress/archivers/zip/ExtraFieldUtils.java
+++ b/src/main/java/org/apache/commons/compress/archivers/zip/ExtraFieldUtils.java
@@ -194,11 +194,12 @@ public ZipExtraField fill(final ZipExtraField field, final byte[] data, final in
         final List<ZipExtraField> v = new ArrayList<>();
         int start = 0;
         LOOP:
+        int dataLength = data.length;
         while (start <= data.length - WORD) {
             final ZipShort headerId = new ZipShort(data, start);
             final int length = new ZipShort(data, start + 2).getValue();
-            if (start + WORD + length > data.length) {
-                final ZipExtraField field = parsingBehavior.onUnparseableExtraField(data, start, data.length - start,
+            if (start + WORD + length > dataLength) {
+                final ZipExtraField field = parsingBehavior.onUnparseableExtraField(data, start, dataLength - start,
                     local, length);
                 if (field != null) {
                     v.add(field);
