diff --git a/src/main/java/org/apache/commons/math3/analysis/interpolation/AkimaSplineInterpolator.java b/src/main/java/org/apache/commons/math3/analysis/interpolation/AkimaSplineInterpolator.java
index 260cbfb..a538145 100644
--- a/src/main/java/org/apache/commons/math3/analysis/interpolation/AkimaSplineInterpolator.java
+++ b/src/main/java/org/apache/commons/math3/analysis/interpolation/AkimaSplineInterpolator.java
@@ -210,11 +210,12 @@
             double w2 = w * w;
             coefficients[0] = yvals[i];
             coefficients[1] = firstDerivatives[i];
-            coefficients[2] = (3 * (yvals[i + 1] - yvals[i]) / w - 2 *
-                               firstDerivatives[i] - firstDerivatives[i + 1]) /
+            double fdP = firstDerivatives[i + 1];
+			coefficients[2] = (3 * (yvals[i + 1] - yvals[i]) / w - 2 *
+                               firstDerivatives[i] - fdP) /
                               w;
             coefficients[3] = (2 * (yvals[i] - yvals[i + 1]) / w +
-                               firstDerivatives[i] + firstDerivatives[i + 1]) /
+                               firstDerivatives[i] + fdP) /
                               w2;
             polynomials[i] = new PolynomialFunction(coefficients);
         }
