diff --git a/src/main/java/org/apache/commons/codec/language/Metaphone.java b/src/main/java/org/apache/commons/codec/language/Metaphone.java
index a319621..6d1b83f 100644
--- a/src/main/java/org/apache/commons/codec/language/Metaphone.java
+++ b/src/main/java/org/apache/commons/codec/language/Metaphone.java
@@ -91,11 +91,12 @@
      */
     public String metaphone(final String txt) {
         boolean hard = false;
-        if (txt == null || txt.length() == 0) {
+        int txtLength = txt.length();
+		if (txt == null || txtLength == 0) {
             return "";
         }
         // single character is itself
-        if (txt.length() == 1) {
+        if (txtLength == 1) {
             return txt.toUpperCase(java.util.Locale.ENGLISH);
         }
 
