diff --git a/oak-search/src/main/java/org/apache/jackrabbit/oak/plugins/index/search/util/FunctionIndexProcessor.java b/oak-search/src/main/java/org/apache/jackrabbit/oak/plugins/index/search/util/FunctionIndexProcessor.java
index 8e24f4a..d988260 100644
--- a/oak-search/src/main/java/org/apache/jackrabbit/oak/plugins/index/search/util/FunctionIndexProcessor.java
+++ b/oak-search/src/main/java/org/apache/jackrabbit/oak/plugins/index/search/util/FunctionIndexProcessor.java
@@ -233,7 +233,8 @@
             }
             prop = prop.substring(0, prop.lastIndexOf(']'));
             remaining = remaining.substring(prop.length() + 1);
-            return property(prop.replaceAll("]]", "]"));
+            String x = prop.replaceAll("]]", "]");
+			return property(x);
         } else {
             String prop = remaining;
             int paren = remaining.indexOf(')');
