diff --git a/oak-jcr/src/main/java/org/apache/jackrabbit/oak/jcr/PropertyDelegate.java b/oak-jcr/src/main/java/org/apache/jackrabbit/oak/jcr/PropertyDelegate.java
index 1e53df9..5f06daf 100644
--- a/oak-jcr/src/main/java/org/apache/jackrabbit/oak/jcr/PropertyDelegate.java
+++ b/oak-jcr/src/main/java/org/apache/jackrabbit/oak/jcr/PropertyDelegate.java
@@ -68,7 +68,8 @@
 
     @Override
     public String getPath() throws InvalidItemStateException {
-        return getParent().getPath() + '/' + getName();
+        String path = getParent().getPath();
+		return path + '/' + getName();
     }
 
     @Override
