diff --git a/nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/state/providers/zookeeper/ZooKeeperStateProvider.java b/nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/state/providers/zookeeper/ZooKeeperStateProvider.java
index 5c8b4c4eda..8fe604b2e5 100644
--- a/nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/state/providers/zookeeper/ZooKeeperStateProvider.java
+++ b/nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/state/providers/zookeeper/ZooKeeperStateProvider.java
@@ -244,10 +244,11 @@ public class ZooKeeperStateProvider extends AbstractStateProvider {
             ZKUtil.deleteRecursive(getZooKeeper(), getComponentPath(componentId));
         } catch (final KeeperException ke) {
             // Node doesn't exist so just ignore
-            if (Code.NONODE == ke.code()) {
+            Code exceptionCode = ke.code();
+            if (Code.NONODE == exceptionCode) {
                 return;
             }
-            if (Code.SESSIONEXPIRED == ke.code()) {
+            if (Code.SESSIONEXPIRED == exceptionCode) {
                 invalidateClient();
                 onComponentRemoved(componentId);
                 return;
