diff --git a/nifi-nar-bundles/nifi-standard-bundle/nifi-standard-content-viewer/src/main/java/org/apache/nifi/web/StandardContentViewerController.java b/nifi-nar-bundles/nifi-standard-bundle/nifi-standard-content-viewer/src/main/java/org/apache/nifi/web/StandardContentViewerController.java
index a85c450..92eeb11 100644
--- a/nifi-nar-bundles/nifi-standard-bundle/nifi-standard-content-viewer/src/main/java/org/apache/nifi/web/StandardContentViewerController.java
+++ b/nifi-nar-bundles/nifi-standard-bundle/nifi-standard-content-viewer/src/main/java/org/apache/nifi/web/StandardContentViewerController.java
@@ -50,19 +50,20 @@
         final ViewableContent content = (ViewableContent) request.getAttribute(ViewableContent.CONTENT_REQUEST_ATTRIBUTE);
 
         // handle json/xml
-        if ("application/json".equals(content.getContentType()) || "application/xml".equals(content.getContentType()) || "text/plain".equals(content.getContentType())) {
+        String contentType = content.getContentType();
+		if ("application/json".equals(contentType) || "application/xml".equals(contentType) || "text/plain".equals(contentType)) {
             final String formatted;
 
             // leave the content alone if specified
             if (DisplayMode.Original.equals(content.getDisplayMode())) {
                 formatted = content.getContent();
             } else {
-                if ("application/json".equals(content.getContentType())) {
+                if ("application/json".equals(contentType)) {
                     // format json
                     final ObjectMapper mapper = new ObjectMapper();
                     final Object objectJson = mapper.readValue(content.getContentStream(), Object.class);
                     formatted = mapper.writerWithDefaultPrettyPrinter().writeValueAsString(objectJson);
-                } else if ("application/xml".equals(content.getContentType())) {
+                } else if ("application/xml".equals(contentType)) {
                     // format xml
                     final StringWriter writer = new StringWriter();
 
@@ -89,12 +90,12 @@
             }
 
             // defer to the jsp
-            request.setAttribute("mode", content.getContentType());
+            request.setAttribute("mode", contentType);
             request.setAttribute("content", formatted);
             request.getRequestDispatcher("/WEB-INF/jsp/codemirror.jsp").include(request, response);
         } else {
             final PrintWriter out = response.getWriter();
-            out.println("Unexpected content type: " + content.getContentType());
+            out.println("Unexpected content type: " + contentType);
         }
     }
 }
