diff --git a/oak-core/src/main/java/org/apache/jackrabbit/oak/query/QueryImpl.java b/oak-core/src/main/java/org/apache/jackrabbit/oak/query/QueryImpl.java
index 2e45555..dc4c1a7 100644
--- a/oak-core/src/main/java/org/apache/jackrabbit/oak/query/QueryImpl.java
+++ b/oak-core/src/main/java/org/apache/jackrabbit/oak/query/QueryImpl.java
@@ -963,11 +963,12 @@
             } else {
                 cost = index.getCost(filter, rootState);
             }
-            if (LOG.isDebugEnabled()) {
-                logDebug("cost for " + index.getIndexName() + " is " + cost);
+            String indexName = index.getIndexName();
+			if (LOG.isDebugEnabled()) {
+                logDebug("cost for " + indexName + " is " + cost);
             }
             if (cost < 0) {
-                LOG.error("cost below 0 for " + index.getIndexName() + " is " + cost);
+                LOG.error("cost below 0 for " + indexName + " is " + cost);
             }
             if (cost < bestCost) {
                 bestCost = cost;
