diff --git a/oak-core/src/main/java/org/apache/jackrabbit/oak/security/user/UserImpl.java b/oak-core/src/main/java/org/apache/jackrabbit/oak/security/user/UserImpl.java
index aa81d37..c0a5138 100644
--- a/oak-core/src/main/java/org/apache/jackrabbit/oak/security/user/UserImpl.java
+++ b/oak-core/src/main/java/org/apache/jackrabbit/oak/security/user/UserImpl.java
@@ -25,6 +25,7 @@
 import org.apache.jackrabbit.api.security.user.User;
 import org.apache.jackrabbit.oak.api.PropertyState;
 import org.apache.jackrabbit.oak.api.Tree;
+import org.apache.jackrabbit.oak.namepath.NamePathMapper;
 import org.apache.jackrabbit.oak.spi.security.user.AuthorizableType;
 import org.apache.jackrabbit.oak.spi.security.user.UserConstants;
 import org.apache.jackrabbit.oak.spi.security.user.util.PasswordUtil;
@@ -64,10 +65,11 @@
     public Principal getPrincipal() throws RepositoryException {
         Tree userTree = getTree();
         String principalName = getPrincipalName();
-        if (isAdmin()) {
-            return new AdminPrincipalImpl(principalName, userTree, getUserManager().getNamePathMapper());
+        NamePathMapper npMapper = getUserManager().getNamePathMapper();
+		if (isAdmin()) {
+            return new AdminPrincipalImpl(principalName, userTree, npMapper);
         } else {
-            return new TreeBasedPrincipal(principalName, userTree, getUserManager().getNamePathMapper());
+            return new TreeBasedPrincipal(principalName, userTree, npMapper);
         }
     }
 
