diff --git a/oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/mongomk/NodeDocument.java b/oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/mongomk/NodeDocument.java
index 3c88a6a..dfd4ee6 100644
--- a/oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/mongomk/NodeDocument.java
+++ b/oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/mongomk/NodeDocument.java
@@ -624,7 +624,8 @@
         if (revisions == null) {
             return false;
         }
-        String value = revisions.get(revision.toString());
+        String r = revision.toString();
+		String value = revisions.get(r);
         if (value == null) {
             return false;
         }
