diff --git a/gremlin-core/src/main/java/com/tinkerpop/gremlin/process/graph/step/branch/RepeatStep.java b/gremlin-core/src/main/java/com/tinkerpop/gremlin/process/graph/step/branch/RepeatStep.java
index 2e39b7efaa..af06601a48 100644
--- a/gremlin-core/src/main/java/com/tinkerpop/gremlin/process/graph/step/branch/RepeatStep.java
+++ b/gremlin-core/src/main/java/com/tinkerpop/gremlin/process/graph/step/branch/RepeatStep.java
@@ -31,7 +31,8 @@ public final class RepeatStep<S> extends AbstractStep<S, S> implements PathConsu
         //try {
         this.repeatTraversal = repeatTraversal; // .clone()
         this.repeatTraversal.asAdmin().addStep(new MarkerIdentityStep<>(this.repeatTraversal)); // TODO: this is really bad
-        this.repeatTraversal.asAdmin().mergeSideEffects(this.getTraversal().asAdmin().getSideEffects());
+        Traversal.SideEffects parentSideEffects = this.getTraversal().asAdmin().getSideEffects();
+        this.repeatTraversal.asAdmin().mergeSideEffects(parentSideEffects);
         this.repeatTraversal.asAdmin().setStrategies(this.getTraversal().asAdmin().getStrategies());
         //} catch (final CloneNotSupportedException e) {
         //    throw new IllegalArgumentException(e.getMessage(), e);
