diff --git a/gremlin-core/src/main/java/com/tinkerpop/gremlin/structure/io/graphson/GraphSONModule.java b/gremlin-core/src/main/java/com/tinkerpop/gremlin/structure/io/graphson/GraphSONModule.java
index 0534c216c6..aa6984ad44 100644
--- a/gremlin-core/src/main/java/com/tinkerpop/gremlin/structure/io/graphson/GraphSONModule.java
+++ b/gremlin-core/src/main/java/com/tinkerpop/gremlin/structure/io/graphson/GraphSONModule.java
@@ -83,7 +83,8 @@ public class GraphSONModule extends SimpleModule {
             m.put(GraphSONTokens.LABEL, edge.getLabel());
             m.put(GraphSONTokens.TYPE, GraphSONTokens.EDGE);
             m.put(GraphSONTokens.IN, edge.getVertex(Direction.IN).getId());
-            m.put(GraphSONTokens.OUT, edge.getVertex(Direction.OUT).getId());
+            Vertex outV = edge.getVertex(Direction.OUT);
+            m.put(GraphSONTokens.OUT, outV.getId());
             m.put(GraphSONTokens.PROPERTIES,
                     edge.getProperties().values().stream().collect(
                             Collectors.toMap(Property::getKey, Property::get)));
