diff --git a/johnzon-core/src/main/java/org/apache/johnzon/core/JsonReaderImpl.java b/johnzon-core/src/main/java/org/apache/johnzon/core/JsonReaderImpl.java
index 86d7708..88dba76 100644
--- a/johnzon-core/src/main/java/org/apache/johnzon/core/JsonReaderImpl.java
+++ b/johnzon-core/src/main/java/org/apache/johnzon/core/JsonReaderImpl.java
@@ -25,6 +25,7 @@
 import javax.json.JsonReader;
 import javax.json.JsonStructure;
 import javax.json.stream.JsonParser;
+import javax.json.stream.JsonParser.Event;
 import javax.json.stream.JsonParsingException;
 
 class JsonReaderImpl implements JsonReader {
@@ -43,7 +44,8 @@
         if (!parser.hasNext()) {
             throw new IllegalStateException("Nothing to read");
         }
-        switch (parser.next()) {
+        Event next = parser.next();
+		switch (next) {
             case START_OBJECT:
                 final JsonObjectBuilder objectBuilder = new JsonObjectBuilderImpl();
                 parseObject(objectBuilder);
@@ -62,7 +64,7 @@
                 return arrayBuilder.build();
             default:
                 close();
-                throw new JsonParsingException("Unknown structure: " + parser.next(), parser.getLocation());
+                throw new JsonParsingException("Unknown structure: " + next, parser.getLocation());
         }
 
     }
