diff --git a/oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/document/DocumentNodeStore.java b/oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/document/DocumentNodeStore.java
index fdfc2d2260..83600c6c0e 100644
--- a/oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/document/DocumentNodeStore.java
+++ b/oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/document/DocumentNodeStore.java
@@ -2084,29 +2084,30 @@ public final class DocumentNodeStore
         final long getChildrenDoneIn = debug ? now() : 0;
 
         String diffAlgo;
+        Revision toRev = to.getLastRevision();
         if (!fromChildren.hasMore && !toChildren.hasMore) {
             diffAlgo = "diffFewChildren";
             diffFewChildren(w, from.getPath(), fromChildren,
-                    from.getLastRevision(), toChildren, to.getLastRevision());
+                    from.getLastRevision(), toChildren, toRev);
         } else {
             if (FAST_DIFF) {
                 diffAlgo = "diffManyChildren";
                 diffManyChildren(w, from.getPath(),
-                        from.getLastRevision(), to.getLastRevision());
+                        from.getLastRevision(), toRev);
             } else {
                 diffAlgo = "diffAllChildren";
                 max = Integer.MAX_VALUE;
                 fromChildren = getChildren(from, null, max);
                 toChildren = getChildren(to, null, max);
                 diffFewChildren(w, from.getPath(), fromChildren,
-                        from.getLastRevision(), toChildren, to.getLastRevision());
+                        from.getLastRevision(), toChildren, toRev);
             }
         }
 
         if (debug) {
             long end = now();
             LOG.debug("Diff performed via '{}' at [{}] between revisions [{}] => [{}] took {} ms ({} ms)",
-                    diffAlgo, from.getPath(), from.getLastRevision(), to.getLastRevision(),
+                    diffAlgo, from.getPath(), from.getLastRevision(), toRev,
                     end - start, getChildrenDoneIn - start);
         }
         return w.toString();
