diff --git a/oak-solr-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/solr/index/SolrIndexEditor.java b/oak-solr-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/solr/index/SolrIndexEditor.java
index b53877f7e1..89eea9b966 100644
--- a/oak-solr-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/solr/index/SolrIndexEditor.java
+++ b/oak-solr-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/solr/index/SolrIndexEditor.java
@@ -240,8 +240,9 @@ class SolrIndexEditor implements IndexEditor {
                             inputDocument.addField(property.getName(), s);
                         }
                     } else {
+                        String value = property.getValue(Type.STRING);
                         inputDocument.addField(
-                                property.getName(), property.getValue(Type.STRING));
+                                property.getName(), value);
                     }
                 }
             }
