diff --git a/oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/segment/file/TarWriter.java b/oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/segment/file/TarWriter.java
index 800f0a7eb9..9a07ede45d 100644
--- a/oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/segment/file/TarWriter.java
+++ b/oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/segment/file/TarWriter.java
@@ -167,9 +167,10 @@ class TarWriter {
             int refcount = segment.get(pos + REF_COUNT_OFFSET) & 0xff;
             int refend = pos + 16 * (refcount + 1);
             for (int refpos = pos + 16; refpos < refend; refpos += 16) {
-                references.add(new UUID(
+                UUID refid = new UUID(
                         segment.getLong(refpos),
-                        segment.getLong(refpos + 8)));
+                        segment.getLong(refpos + 8));
+                references.add(refid);
             }
         }
 
