diff --git a/oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/nodetype/write/NodeTypeTemplateImpl.java b/oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/nodetype/write/NodeTypeTemplateImpl.java
index 0e3db4d148..a2f137b5ae 100644
--- a/oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/nodetype/write/NodeTypeTemplateImpl.java
+++ b/oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/nodetype/write/NodeTypeTemplateImpl.java
@@ -177,7 +177,8 @@ class NodeTypeTemplateImpl extends NamedTemplate
         if (nodeDefinitionTemplates != null) {
             int ndn = 1;
             for (NodeDefinitionTemplateImpl ndt : nodeDefinitionTemplates) {
-                Tree tree = type.addChild(JCR_CHILDNODEDEFINITION + "[" + ndn++ + "]");
+                String name = JCR_CHILDNODEDEFINITION + "[" + ndn++ + "]";
+                Tree tree = type.addChild(name);
                 tree.setProperty(
                         JCR_PRIMARYTYPE, NT_CHILDNODEDEFINITION, Type.NAME);
                 ndt.writeTo(tree);
