diff --git a/oak-core/src/main/java/org/apache/jackrabbit/oak/security/authorization/AccessControlManagerImpl.java b/oak-core/src/main/java/org/apache/jackrabbit/oak/security/authorization/AccessControlManagerImpl.java
index f34a997fa3..f98bc58a34 100644
--- a/oak-core/src/main/java/org/apache/jackrabbit/oak/security/authorization/AccessControlManagerImpl.java
+++ b/oak-core/src/main/java/org/apache/jackrabbit/oak/security/authorization/AccessControlManagerImpl.java
@@ -227,8 +227,9 @@ public class AccessControlManagerImpl implements JackrabbitAccessControlManager,
 
             ACL acl = (ACL) policy;
             for (JackrabbitAccessControlEntry ace : acl.getACEs()) {
-                String nodeName = generateAceName(aclNode, ace.isAllow());
-                String ntName = (ace.isAllow()) ? NT_REP_GRANT_ACE : NT_REP_DENY_ACE;
+                boolean isAllow = ace.isAllow();
+                String nodeName = generateAceName(aclNode, isAllow);
+                String ntName = isAllow ? NT_REP_GRANT_ACE : NT_REP_DENY_ACE;
 
                 NodeUtil aceNode = aclNode.addChild(nodeName, ntName);
                 aceNode.setString(REP_PRINCIPAL_NAME, ace.getPrincipal().getName());
