diff --git a/oak-lucene/src/main/java/org/apache/jackrabbit/oak/plugins/index/lucene/LuceneDocumentMaker.java b/oak-lucene/src/main/java/org/apache/jackrabbit/oak/plugins/index/lucene/LuceneDocumentMaker.java
index 77ce9dfda3..2044f100b6 100644
--- a/oak-lucene/src/main/java/org/apache/jackrabbit/oak/plugins/index/lucene/LuceneDocumentMaker.java
+++ b/oak-lucene/src/main/java/org/apache/jackrabbit/oak/plugins/index/lucene/LuceneDocumentMaker.java
@@ -347,7 +347,8 @@ public class LuceneDocumentMaker extends FulltextDocumentMaker<Document> {
         boolean added = false;
         for (String nodeName : propertNode.getChildNodeNames()) {
             NodeState dynaTag = propertNode.getChildNode(nodeName);
-            String dynaTagName = dynaTag.getProperty(DYNAMIC_BOOST_TAG_NAME).getValue(Type.STRING);
+            PropertyState p = (PropertyState) dynaTag.getProperty(DYNAMIC_BOOST_TAG_NAME);
+            String dynaTagName = p.getValue(Type.STRING);
             Double dynaTagConfidence = dynaTag.getProperty(DYNAMIC_BOOST_TAG_CONFIDENCE).getValue(Type.DOUBLE);
 
             List<String> tokens = new ArrayList<>(splitForIndexing(dynaTagName));
