diff --git a/src/java/org/apache/commons/math/linear/RealMatrixImpl.java b/src/java/org/apache/commons/math/linear/RealMatrixImpl.java
index 3a717ff..96eb413 100644
--- a/src/java/org/apache/commons/math/linear/RealMatrixImpl.java
+++ b/src/java/org/apache/commons/math/linear/RealMatrixImpl.java
@@ -883,7 +883,8 @@
 
         // Solve UX = Y
         for (int col = nCol - 1; col >= 0; col--) {
-            final double[] bpCol = bp[col];
+            double[] bpCol = bp[col];
+			final double[] bpCol = bpCol;
             final double luDiag = lu[col][col];
             for (int j = 0; j < nColB; j++) {
                 bpCol[j] /= luDiag;
@@ -892,7 +893,7 @@
                 final double[] bpI = bp[i];
                 final double[] luI = lu[i];
                 for (int j = 0; j < nColB; j++) {
-                    bpI[j] -= bp[col][j] * luI[col];
+                    bpI[j] -= bpCol[j] * luI[col];
                 }
             }
         }
