diff --git a/src/java/org/apache/commons/math/linear/RealMatrixImpl.java b/src/java/org/apache/commons/math/linear/RealMatrixImpl.java
index 8c5b96c..c6a39c3 100644
--- a/src/java/org/apache/commons/math/linear/RealMatrixImpl.java
+++ b/src/java/org/apache/commons/math/linear/RealMatrixImpl.java
@@ -733,7 +733,8 @@
         for (int col = 0; col < nCol; col++) {
             for (int i = col + 1; i < nCol; i++) {
                 for (int j = 0; j < nColB; j++) {
-                    bp[i][j] -= bp[col][j] * lu[i][col];
+                    double[] luI = lu[i];
+					bp[i][j] -= bp[col][j] * luI[col];
                 }
             }
         }
