diff --git a/src/main/java/org/apache/commons/compress/archivers/sevenz/SevenZMethod.java b/src/main/java/org/apache/commons/compress/archivers/sevenz/SevenZMethod.java
index 3c446cc..b2abb3f 100644
--- a/src/main/java/org/apache/commons/compress/archivers/sevenz/SevenZMethod.java
+++ b/src/main/java/org/apache/commons/compress/archivers/sevenz/SevenZMethod.java
@@ -97,8 +97,9 @@
     }
 
     byte[] getId() {
-        final byte[] copy = new byte[id.length];
-        System.arraycopy(id, 0, copy, 0, id.length);
+        int idLength = id.length;
+		final byte[] copy = new byte[idLength];
+        System.arraycopy(id, 0, copy, 0, idLength);
         return copy;
     }
 
