diff --git a/src/main/java/org/apache/commons/compress/compressors/bzip2/BZip2CompressorInputStream.java b/src/main/java/org/apache/commons/compress/compressors/bzip2/BZip2CompressorInputStream.java
index 1de731d..5a175c3 100644
--- a/src/main/java/org/apache/commons/compress/compressors/bzip2/BZip2CompressorInputStream.java
+++ b/src/main/java/org/apache/commons/compress/compressors/bzip2/BZip2CompressorInputStream.java
@@ -549,7 +549,8 @@
                         }
                         groupPos--;
                         zt = selector[groupNo];
-                        zn = minLens[zt];
+                        int minLens_zt = minLens[zt];
+						zn = minLens_zt;
                         zvec = bsR(zn);
                         while (zvec > limit[zt][zn]) {
                             zn++;
