diff --git a/src/main/java/org/joda/time/tz/ZoneInfoCompiler.java b/src/main/java/org/joda/time/tz/ZoneInfoCompiler.java
index b03549f..0aab057 100644
--- a/src/main/java/org/joda/time/tz/ZoneInfoCompiler.java
+++ b/src/main/java/org/joda/time/tz/ZoneInfoCompiler.java
@@ -506,7 +506,8 @@
                 } else if (token.equalsIgnoreCase("Zone")) {
                     zone = new Zone(st);
                 } else if (token.equalsIgnoreCase("Link")) {
-                    iLinks.add(st.nextToken());
+                    String real = st.nextToken();
+					iLinks.add(real);
                     iLinks.add(st.nextToken());
                 } else {
                     System.out.println("Unknown line: " + line);
