diff --git a/src/main/java/org/joda/time/tz/ZoneInfoCompiler.java b/src/main/java/org/joda/time/tz/ZoneInfoCompiler.java
index b03549f..256f006 100644
--- a/src/main/java/org/joda/time/tz/ZoneInfoCompiler.java
+++ b/src/main/java/org/joda/time/tz/ZoneInfoCompiler.java
@@ -507,7 +507,8 @@
                     zone = new Zone(st);
                 } else if (token.equalsIgnoreCase("Link")) {
                     iLinks.add(st.nextToken());
-                    iLinks.add(st.nextToken());
+                    String alias = st.nextToken();
+					iLinks.add(alias);
                 } else {
                     System.out.println("Unknown line: " + line);
                 }
